main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -740,12 +728,25 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       Static | Scope | LetMatch | LetMatchStar | LetRecDef | LetUse | LetImport | IfThenElse |
         Choose | TypeMatch | Match | LambdaMatch | Try | Lambda | Tuple |
         RecordOperation | RecordLiteral | Block | RecordSelectLambda |
-        SelectChannel | Spawn | ParYield | Par | Lazy | Force | Upcast | Supercast | Mask | Intrinsic | New | ArrayLit | ArrayNew |
+        SelectChannel | Spawn | ParYield | Par | Lazy | Force | Cast |
+        Upcast | Supercast | Mask | Intrinsic | New | ArrayLit | ArrayNew |
         FNil | FSet | FMap | ConstraintSet | FixpointLambda | FixpointProject | FixpointSolveWithProject |
         FixpointQueryWithSelect | ConstraintSingleton | Interpolation | Literal | Resume | Do |
         Discard | Debug | ForYield | ForEach | NewObject | UnaryLambda | FName | Tag | Hole
     }
 
+    def Cast: Rule1[ParsedAst.Expression] = rule {
+      SP ~ keyword("unsafe_cast") ~ WS ~ Expression ~ optWS ~ TypAndPurFragment ~ SP ~> ParsedAst.Expression.Cast
```

Comment:
Don't you need a `:` here?

Reply:
Yes, sorry this version was just an experiment. See https://github.com/flix/flix/pull/4958/commits/248132beb09867ed3172a2f0b309872ea91ea9f1

Reply:
You need to pick an expression after Ascribe.

Reply:
Instead of `keyword("unsafe_cast") ~ WS ~ Expression` you want `keyword("unsafe_cast") ~ WS ~ Expression.FAppend` ...

Reply:
Doesn't work either:
```scala
    def Cast: Rule1[ParsedAst.Expression] = rule {
      SP ~ keyword("unsafe_cast") ~ WS ~ FAppend ~ optWS ~ TypAndPurFragment ~ SP ~> ParsedAst.Expression.Cast
    }
```

```

Unable to compile. Failed with: 23 errors.

-- Parse Error -------------------------------------------------- MutDeque.flix

>> Parse Error: Invalid input ": ", expected "->", ArgumentList, '[', '.', ::, ::: or TypAndPurFragment (line 39, column 68):
    pub def regionOf(_: MutDeque[a, r]): Region[r] = unsafe_cast (): Region[r]
                                                                   ^
```

Reply:
Yes but it still fails
```scala
    def Cast: Rule1[ParsedAst.Expression] = rule {
      SP ~ keyword("unsafe_cast") ~ WS ~ FAppend ~ optWS ~ ":" ~ optWS ~ TypAndPurFragment ~ SP ~> ParsedAst.Expression.Cast
    }

```
```
Unable to compile. Failed with: 2 errors.

-- Parse Error -------------------------------------------------- Prelude.flix

>> Parse Error: Invalid input '@', expected BinaryTail, "without", Op, '`', <+>, **, *, /, mod, rem, +, -, <<<, >>>, <=, >=, <, > or == (line 170, column 1):
@Internal
^

-- Parse Error -------------------------------------------------- Nel.flix

>> Parse Error: Invalid input ')', expected BinaryTail, Op, '`', <+>, **, *, /, mod, rem, +, -, <<<, >>>, ==, !=, <=>, &&&, ^^^ or ||| (line 403, column 76):
        loop(x :: xs, unsafe_cast ks -> ks: & (ef1 and ef2) \ (ef1_ + ef2_))
                                                                           ^
```
The `@Internal` comes after an unsafe cast. Again it seems like it expects more input somehow. Is it correct that `FAppend` can derived as part of a primary expression?

Reply:
Hmmm

Reply:
I see the problem. But not yet a solution.

Reply:
I will keep trying out ideas meanwhile

Reply:
Can we do `unsafe_cast exp as Type` as a temporary solution? Does that work?